src/test/run-pass/never-type-args.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```

Comment:
There should be a MIR codegen test too. You can find some examples in `src/test/mir-opt`.

Reply:
I've added one — I wasn't 100% sure which MIR pass to check after, so I've gone for `SimplifyCfg-initial` for now. I just tried to check that the right basic blocks were being marked as unreachable — let me know if this isn't quite stringent enough.

Reply:
`SimplifyCfg-initial` is okay.